### ISSUE TITLE
Questions about data input

### ISSUE THREAD

**Issue by veloseptimus** [kychern/FacialRetargeting#9]
This is a beginner of deep learning and needs hint about data input.
 I could run the program well with your test data, but have problems using my own .obj models.
Is there any restrictions for the input models of RIMD?Is there any parameters I should modify as the input changes?(vertice numbers?)
I would really appreciate it if you answered.

**Comment by kychern** [comment]
Hi, for RIMD feature computing, the reference mesh and deformed mesh must be in the same topology (vertices and faces). If you want to calculate it for your own models, please make sure to modify the correct reference model path.

**Comment by veloseptimus** [comment]
Thank you for replying.
Thanks for your help, I've got the DR feature computed. But it turned error using the pretrained model, with details below.
"Dimensions must be equal, but are 11510 and 468 for 'lambda_1/SparseTensorDenseMatMul/SparseTensorDenseMatMul' (op: 'SparseTensorDenseMatMul') with input shapes: [11510,2], [11510], [2], [468,9] and with input tensors computed as partial shapes: input[2] = [11510,11510]."
I have set input dimension 468x9 in main.py as my face mesh has 468 vertices. Do you know where might go wrong?

**Comment by veloseptimus** [comment]
Thank you for answering.
Thanks to your help, I have computed the RIMD feature. But the pose of the reconstruction mesh seems wrong, do I have to make the deformation mesh aligned?
And the pretrained model doesn't seems compatible with my own input face mesh. Do I have to train my own model?

**Comment by kychern** [comment]
> Thank you for answering. Thanks to your help, I have computed the RIMD feature. But the pose of the reconstruction mesh seems wrong, do I have to make the deformation mesh aligned?

Yes, spatial alignment is necessary to obtain the reconstruction mesh. You may refer to [](https://github.com/zihangJiang/DR-Learning-for-3D-Face/issues/12) to find out more on this.
